MariaDB 10.3.8

Mno-hime opened the pull request:
Forked off MariaDB 10.1 component.

Tested: DB start, password change, mediator version change (from 10.1),
retrieve info from the DB (ping, variables, version).

Test results are sorted because `mtr` does not provide stable order of
tests for comparison.

pyhalov commented on components/database/mariadb-103/Makefile:
```diff
@@ -0,0 +1,142 @@
+REQUIRED_PACKAGES += $(GXX_RUNTIME_PKG)
+REQUIRED_PACKAGES += compress/bzip2
+REQUIRED_PACKAGES += compress/xz
+REQUIRED_PACKAGES += database/mariadb-103/client
```
please, remove dependency on itself, it shouldn't be required.

pyhalov commented on components/database/mariadb-103/files/mariadb_103:
```diff
@@ -0,0 +1,96 @@
+		exit $SMF_EXIT_ERR_CONFIG
+		else
+		echo ${MYSQLBIN}/64/mysqld_safe --user=mysql --datadir=${MYSQLDATA} --pid-file=${PIDFILE}
+		${MYSQLBIN}/64/mysqld_safe --user=mysql --datadir=${MYSQLDATA} --pid-file=${PIDFILE} > /dev/null &
```
Should we do 64-bit the default one? Or perhaps, stop shipping 32-bit server at all?

Mno-hime replied:
My take is to build the whole component 64-bit-only. I guess we should not promote 32-bit in new components. What do you think?

pyhalov replied:
It would be better to deliver 32-bit libraries (when we change default mysql implementation, this shouldn't be an issue.

Mno-hime committed:
MariaDB 10.3.8

Forked off MariaDB 10.1 component, server is 64-bit-only, libraries also
32-bit.

Tested: DB start, password change, mediator version change (from 10.1),
retrieve info from the DB (ping, variables, version).

Test results are sorted because `mtr` does not provide stable order of
tests for comparison.

pyhalov commented on components/database/mariadb-103/Makefile:
```diff
@@ -0,0 +1,138 @@
+CMAKE_OPTIONS+= -DCMAKE_REQUIRED_INCLUDES=/usr/include/pcre
+CMAKE_OPTIONS+= -DWITH_INNOBASE_STORAGE_ENGINE=1
+CMAKE_OPTIONS+= -DWITH_INNODB_MEMCACHED=ON
+CMAKE_OPTIONS+= -DWITH_INNODB_DISALLOW_WRITES=1
```
Why do you turn it on?

Mno-hime commented:
Currently can't build 32-bit due to:
```
-- SYSTEM_LIBS nsl;socket;/usr/lib/libz.so;socket;m;m;/usr/lib/libssl.so;/usr/lib/libcrypto.so
CMake Error at libmariadb/cmake/install_plugins.cmake:11 (INSTALL):
  INSTALL TARGETS given no LIBRARY DESTINATION for module target "dialog".
Call Stack (most recent call first):
  libmariadb/cmake/plugins.cmake:82 (INSTALL_PLUGIN)
  libmariadb/plugins/auth/CMakeLists.txt:14 (REGISTER_PLUGIN)
  libmariadb/plugins/CMakeLists.txt:7 (INCLUDE)
  libmariadb/CMakeLists.txt:371 (INCLUDE)


CMake Error at libmariadb/cmake/install_plugins.cmake:11 (INSTALL):
  INSTALL TARGETS given no LIBRARY DESTINATION for module target
  "auth_gssapi_client".
Call Stack (most recent call first):
  libmariadb/cmake/plugins.cmake:82 (INSTALL_PLUGIN)
  libmariadb/plugins/auth/CMakeLists.txt:32 (REGISTER_PLUGIN)
  libmariadb/plugins/CMakeLists.txt:7 (INCLUDE)
  libmariadb/CMakeLists.txt:371 (INCLUDE)


CMake Error at libmariadb/cmake/install_plugins.cmake:11 (INSTALL):
  INSTALL TARGETS given no LIBRARY DESTINATION for module target
  "sha256_password".
Call Stack (most recent call first):
  libmariadb/cmake/plugins.cmake:82 (INSTALL_PLUGIN)
  libmariadb/plugins/auth/CMakeLists.txt:47 (REGISTER_PLUGIN)
  libmariadb/plugins/CMakeLists.txt:7 (INCLUDE)
  libmariadb/CMakeLists.txt:371 (INCLUDE)


CMake Error at libmariadb/cmake/install_plugins.cmake:11 (INSTALL):
  INSTALL TARGETS given no LIBRARY DESTINATION for module target
  "mysql_clear_password".
Call Stack (most recent call first):
  libmariadb/cmake/plugins.cmake:82 (INSTALL_PLUGIN)
  libmariadb/plugins/auth/CMakeLists.txt:64 (REGISTER_PLUGIN)
  libmariadb/plugins/CMakeLists.txt:7 (INCLUDE)
  libmariadb/CMakeLists.txt:371 (INCLUDE)


-- Dynamic column API support: ON
SYSTEM processor: i386
-- MariaDB Connector/c configuration:
-- Static PLUGINS mysql_native_password;mysql_old_password;pvio_socket
-- Dynamic PLUGINS dialog;auth_gssapi_client;sha256_password;mysql_clear_password
-- CPack generation: TGZ
-- SSL support: OPENSSL Libs: /usr/lib/libssl.so;/usr/lib/libcrypto.so
-- Zlib support: ON
-- Installation layout: DEFAULT
-- Include files will be installed in include
-- Libraries will be installed in lib
-- Binaries will be installed in bin/i86
-- Documentation included from 
-- Required: 
CMake Error at cmake/install_macros.cmake:243 (MESSAGE):
  Need DESTINATION parameter for MYSQL_INSTALL_TARGETS
Call Stack (most recent call first):
  cmake/plugin.cmake:251 (MYSQL_INSTALL_TARGETS)
  storage/archive/CMakeLists.txt:17 (MYSQL_ADD_PLUGIN)
```
64-bit seems to build.

Ideas?

Mno-hime commented:
Nah, `CMAKE_PLUGINDIR.32` wasn't set.

pyhalov commented on components/database/mariadb-103/Makefile:
```diff
@@ -0,0 +1,138 @@
+CMAKE_OPTIONS+= -DCMAKE_CXX_COMPILER=$(CXX)
+CMAKE_OPTIONS+= -DCMAKE_C_FLAGS=$(CFLAGS)
+# `-fpermissive` due to:
+# 	sql/sql_class.h:6416:42: warning: invalid conversion from 'const char*' to 'char*' [-fpermissive]
```
Should we add patch to use explicit conversion?

Mno-hime replied:
Workaround removed, for changes see `patches/08-dtrace-invalid-conversion.patch`.

pyhalov commented on components/database/mariadb-103/mariadb-103.p5m:
```diff
@@ -0,0 +1,247 @@
+    path=lib/svc/manifest/application/database/mariadb_103.xml
+file files/mariadb_103 path=lib/svc/method/mariadb_103 mode=0555
+
+link path=etc/mysql/my.cnf target=../mariadb/10.3/my.cnf mediator=mysql mediator-version=10.3 mediator-implementation=mariadb
```
If we want to continue shipping MariaDB 10.1 by default (and we likely do), it's worth to add 
```
<transform link mediator=mysql  mediator-implementation=mariadb mediator-version=10.1 -> default mediator-priority vendor>
``` 
to transforms/defaults and bump MariaDB 10.1 COMPONENT_REVISION

Mno-hime replied:
See https://github.com/OpenIndiana/oi-userland/pull/4361.
